src/test/java/dawahdevs/qa/api/controllerTest/QaApiControllerTest.java

```diff
@@ -29,15 +35,65 @@
 	private MockMvc mockMvc;
 
 	@MockBean
-	private QuestionManager qaApiManager;
+	private QaApiManager qaApiManager;
 
 	@Before
 	public void init() {
 		initMocks(this);
 	}
 
 	@Test
-	public void testGetAllStatuses() {
+	public void testGetQuestions() throws Exception {
+		when(qaApiManager.getQuestions()).thenReturn(new AllQuestionsResponse());
+
+		mockMvc.perform(get("/v1/questions")).andExpect(status().isOk()).andDo(print());
+
+		verify(qaApiManager, times(1)).getQuestions();
+	}
+
+	@Test
+	public void testGetQuestionById() throws Exception {
+		when(qaApiManager.getQuestionById(1)).thenReturn(new QuestionByIdResponse());
+
+		mockMvc.perform(get("/v1/questions/1")).andExpect(status().isOk()).andDo(print());
+
+		verify(qaApiManager, times(1)).getQuestionById(1);
```

Comment:
aren't you suppose to mock the `QuestionRepository` object only. I mean this doesn't add much testing value. This test is currently mocking the manager method then checking if it was mocked correctly. You should only mock what you cant control like a fetch request, an ORM, a repository query, third party libraries, etc. 

> same applies to the other tests

Reply:
So I definitely tested many different ways... one being going all the way to the "DAO" and making sure I got there. HOWEVER this way as this is according to https://tanzu.vmware.com/application-modernization-recipes/testing/spring-boot-testing-best-practices .... You should ONLY test what this class controls so this test mocks a call and makes sure it goes to the proper manager.... and if you think about it thats testing the exact and only functionality the controller controls. You will see that I only test what the Class itself is control over